Elements/src/Geometry/Profiles/ParametricProfile.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Scripting;
+using Microsoft.CodeAnalysis.Scripting;
+using Newtonsoft.Json;
+
+namespace Elements.Geometry.Profiles
+{
+    /// <summary>
+    /// A profile whose vertex locations are defined by a parametric expression.
+    /// </summary>
+    public class ParametricProfile : Profile
+    {
+        private readonly List<MetadataReference> _refs;
+
+        /// <summary>
+        /// The data which defines the profile.
+        /// </summary>
+        public ParametericProfileData Data { get; set; }
+
+        /// <summary>
+        /// Create a parametric profile.
+        /// </summary>
+        /// <param name="data">The profile </param>
+        /// <param name="id">The unique identifier of the profile.</param>
+        /// <param name="name">The name of the profile.</param>
+        [JsonConstructor]
+        public ParametricProfile(ParametericProfileData data, Guid @id = default, string @name = null) : base(null, null, id, name)
+        {
+            if (data.PerimeterVectorExpressions == null || data.PerimeterVectorExpressions.Count == 0)
+            {
+                throw new ArgumentException("The parametric profile could not be created. No translation expressions were provided.");
+            }
+
+            Data = data;
+
+            // These references are required to get dynamic to work.
+            _refs = new List<MetadataReference>{
+                                    MetadataReference.CreateFromFile(typeof(Microsoft.CSharp.RuntimeBinder.RuntimeBinderException).GetTypeInfo().Assembly.Location),
+                                    MetadataReference.CreateFromFile(typeof(System.Runtime.CompilerServices.DynamicAttribute).GetTypeInfo().Assembly.Location)};
+            Perimeter = CreatePolygonFromExpressions(data.PerimeterVectorExpressions, data.PropertyValues, _refs).Result;
```

Comment:
using `.Result` in ctors can/will likely result in deadlocks
https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#constructors